File: back end/payment-service/src/main/java/com/cibertec/service/PaymentStatusService.java
package com.cibertec.service;

import com.cibertec.dto.PaymentStatusRequest;
import com.cibertec.dto.PaymentStatusResponse;

import java.util.List;

public interface PaymentStatusService {
    List<PaymentStatusResponse> getAllPaymentStatuses();
    PaymentStatusResponse getPaymentStatusById(Long id);
    PaymentStatusResponse createPaymentStatus(PaymentStatusRequest paymentStatusRequest);
    PaymentStatusResponse updatePaymentStatus(Long id, PaymentStatusRequest paymentStatusRequest);
    void deletePaymentStatus(Long id);
}

File: la-gula-pizzeria/src/app/shared/services/payment/payment-status-service.ts
import { Injectable } from '@angular/core';
import { GATEWAY_URL, SERVICES_PATHS } from '../config/config';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import PaymentStatusResponse from '../../model/payment/response/paymentStatusResponse.model';
import PaymentStatusRequest from '../../model/payment/request/paymentStatusRequest.model';

@Injectable({
  providedIn: 'root'
})
export class PaymentStatusService {
  path = `${GATEWAY_URL}${SERVICES_PATHS.PAYMENT}` + '/payments-status';

  constructor(private http: HttpClient) { }

  getAllPaymentStatuses() : Observable<PaymentStatusResponse[]> {
    return this.http.get<PaymentStatusResponse[]>(this.path);
  }
  getPaymentStatusById(id: number) : Observable<PaymentStatusResponse> {
    return this.http.get<PaymentStatusResponse>(`${this.path}/${id}`);
  }
  savePaymentStatus(status: PaymentStatusRequest) : Observable<PaymentStatusResponse> {
    return this.http.post<PaymentStatusResponse>(this.path, status);
  }
  updatePaymentStatus(id: number, status: PaymentStatusRequest) : Observable<PaymentStatusResponse> {
    return this.http.put<PaymentStatusResponse>(`${this.path}/${id}`, status);
  }
  deletePaymentStatus(id: number) : Observable<void> {
    return this.http.delete<void>(`${this.path}/${id}`);
  }
}
